### Author: Adriano Yoshino - amy324@nyu.edu

In [5]:
# Code made based on Dr. Bianco notebooks. Majoritary of this code was created by Dr. Bianco.
# This homework was made to reproduct the results obtained in class to another set of data.

import os
import sys
import numpy as np
import pylab as pl


import json
import os

%pylab inline

Populating the interactive namespace from numpy and matplotlib


### Reproducing z-test to "Convicted of a felony after 3 years" data

#### NULL HYPOTHESIS: the % of Recidivism convicted of a felony is the same or higher for candidates who particiated in the program as for the control group. Siginificance level p = 0.05

- $P_1$ is control group and $P_0$ is program participants

- $H_0: P_0 - P_1 \geq$ 0 
- $H_a: P_0 - P_1 $< 0
- $\alpha$ = 0.05

#### this is a TEST OF PROPORTIONS. we use the Binomial distribution since it is a yes/no (bernulli) test for each subject: the former inmate was or was not ever employed in a CEO transitional job (second row in the table above):
$P_0=0.10, P_1=0.117$

In [1]:
alpha=0.05
# we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0 = 10 * 0.01 
P_1 = 11.7 * 0.01

if P_0-P_1 >= 0:
    # we are done
    print ("The Null holds")
else:
    print ("We must assess the statistical significance")

n_0 = 568
n_1 = 409

#lets get the counts by multiplying by the sample size
Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

We must assess the statistical significance


In [10]:
#define the sample proportion first
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

0.107116683726


In [11]:
# I am goonna create a little one line function to calculate the standard error
# and to calculate p

p = lambda p0, p1, n0, n1: (p0 * n0 + p1 * n1) / (n0 + n1)
#standard error
se = lambda p, n0, n1: np.sqrt(p * (1 - p) * (1.0 / n0 + 1.0 / n1)) 

In [12]:
zscore = lambda p0, p1, s : (p0 - p1) / s
z_2y = zscore(P_1, P_0, se(p(P_0, P_1, n_0, n_1), n_0, n_1))
print (z_2y)

0.84764020522


In [2]:
## p-value for recidivism, convicted of a felony (years 1-3):

p_2y = 1 - 0.8023


def report_result(p,a):
    print ('is the p value ' + 
           '{0:.2f} smaller than the critical value {1:.2f}?'.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('The Null hypothesis is {}'.format(\
                            'rejected' if p < a  else 'not rejected') )

    
report_result(p_2y, alpha)

is the p value 0.20 smaller than the critical value 0.05?
NO!
The Null hypothesis is not rejected


### Reproducing  $\chi^2$ test to "Ever employed in a CEO transitional job" data

|employed in subsadized job |     employed          | not employed     |                   
|---------------------------|:---------------------:|------------------|---------------------------|
| test sample               | $0.701*564$           | $0.299*564$      | 564                       |
| control sample            | $0.035*409$           | $0.965*4.09$     | 409                       |
|                           |                       |                  |                           |
| total                     | 409.679               |  562.912         | 973                       |

In [3]:
from evalChisq import evalChisq

In [6]:
Ntot = 973 # a + b + c + d = tot

sample_values = np.array([[0.701 * 564, 0.299 * 564], [0.0305 * 409, 0.965 * 409]])
print (evalChisq(sample_values))

436.223462575


#### For alpha = 0.05,
- 436.2235 is > than 3.84, so the Null Hypothesis is rejected

### Reproducing  $\chi^2$ test to "Convicted of a felony after 3 years" data


|convicted of a fellony     |     yes   | no        |                   
|---------------------------|-----------|-----------|----------------|
| test sample               |$0.10*568$ | $0.90*568$|   568          |
| control sample            |$0.117*409$|$0.883*409$|   409          |
|                           |           |           |                |
| total                     |104.653    |  872.347  |   977          |

In [14]:
from evalChisq import evalChisq

In [15]:
Ntot = 977 # a + b + c + d = tot

sample_values = np.array([[0.10 * 568, 0.9 * 568], [0.117 * 409, 0.883 * 409]])
print (evalChisq(sample_values))

0.718493917505


#### For alpha = 0.05,
- 0.7285 is < than 3.84, so the Null Hypothesis is not rejected